### PR TITLE
Add doc tests for Albert and Bigbird

### PR DESCRIPTION
**Pull request by vumichien**
# What does this PR do?

Add doc tests for Albert and Bigbird, a part of issue https://github.com/huggingface/transformers/issues/16292

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@patrickvonplaten, @ydshieh
Documentation: @sgugger

### CODE REVIEW

**Comment by vumichien** [comment]
@ydshieh Could you please take a look at it? I think we still have a problem with `AlbertTokenizer ` as we have discussed on the Discord channel `the AlbertTokenizer will add an extra "_" just after the "[MASK]" token` which will lead to the different shape between `input_text` and `target_text`. This is the code snippet for checking the output.
```
from transformers import AlbertTokenizer, AlbertForMaskedLM
import torch

tokenizer = AlbertTokenizer.from_pretrained("albert-base-v2")
model = AlbertForMaskedLM.from_pretrained("albert-base-v2")

input_text = "The capital of France is [MASK]."
target_text = "The capital of France is Paris."

tokenizer.tokenize(input_text)
# ['▁the', '▁capital', '▁of', '▁france', '▁is', ' [MASK]', '▁', '.']
tokenizer.tokenize(target_text )
# ['▁the', '▁capital', '▁of', '▁france', '▁is', '▁paris', '.']
```

**Comment by ydshieh** [comment]
Hi, @vumichien 

I won't be available for the next few days. Will check when I am back, or my colleague could check this PR :-) 

Regarding the Albert tokenizer, do you encounter any runtime error due to the shape issue? I understand that the shapes are different,  and had a short discussion with the team. But we thought it should still work. Sorry for not responding this part earlier, but if you see errors due to these shapes, could you post it here, please?

**Comment by vumichien** [comment]
@ydshieh When I run the test for doc, the error will show like the following (sorry for very long error)
```
======================================================================================= FAILURES =======================================================================================
____________________________________________________ [doctest] transformers.models.albert.modeling_albert.AlbertForMaskedLM.forward ____________________________________________________
1034     >>> predicted_token_id = logits[0, mask_token_index].argmax(axis=-1)
1035     >>> tokenizer.decode(predicted_token_id)
1036     'reims'
1037 
1038     ```
1039 
1040     ```python
1041     >>> labels = tokenizer("The capital of France is Paris.", return_tensors="pt")["input_ids"]
1042     >>> # mask labels of non-[MASK] tokens
1043     >>> labels = torch.where(inputs.input_ids == tokenizer.mask_token_id, labels, -100)
UNEXPECTED EXCEPTION: RuntimeError('The size of tensor a (10) must match the size of tensor b (9) at non-singleton dimension 1')
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest transformers.models.albert.modeling_albert.AlbertForMaskedLM.forward[10]>", line 1, in <module>
RuntimeError: The size of tensor a (10) must match the size of tensor b (9) at non-singleton dimension 1
/home/vumichien/Detomo/transformers/src/transformers/models/albert/modeling_albert.py:1043: UnexpectedException
1036     'reims'
1037 
1038     ```
1039 
1040     ```python
1041     >>> labels = tokenizer("The capital of France is Paris.", return_tensors="pt")["input_ids"]
1042     >>> # mask labels of non-[MASK] tokens
1043     >>> labels = torch.where(inputs.input_ids == tokenizer.mask_token_id, labels, -100)
1044 
1045     >>> outputs = model(**inputs, labels=labels)
UNEXPECTED EXCEPTION: ValueError('Expected input batch_size (10) to match target batch_size (9).')
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest transformers.models.albert.modeling_albert.AlbertForMaskedLM.forward[11]>", line 1, in <module>
  File "/home/vumichien/Detomo/transformers/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/vumichien/Detomo/transformers/src/transformers/models/albert/modeling_albert.py", line 964, in forward
    masked_lm_loss = loss_fct(prediction_scores.view(-1, self.config.vocab_size), labels.view(-1))
  File "/home/vumichien/Detomo/transformers/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/vumichien/Detomo/transformers/venv/lib/python3.8/site-packages/torch/nn/modules/loss.py", line 1163, in forward
    return F.cross_entropy(input, target, weight=self.weight,
  File "/home/vumichien/Detomo/transformers/venv/lib/python3.8/site-packages/torch/nn/functional.py", line 2996, in cross_entropy
    return torch._C._nn.cross_entropy_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index, label_smoothing)
ValueError: Expected input batch_size (10) to match target batch_size (9).
/home/vumichien/Detomo/transformers/src/transformers/models/albert/modeling_albert.py:1045: UnexpectedException
1037 
1038     ```
1039 
1040     ```python
1041     >>> labels = tokenizer("The capital of France is Paris.", return_tensors="pt")["input_ids"]
1042     >>> # mask labels of non-[MASK] tokens
1043     >>> labels = torch.where(inputs.input_ids == tokenizer.mask_token_id, labels, -100)
1044 
1045     >>> outputs = model(**inputs, labels=labels)
1046     >>> round(outputs.loss.item(), 2)
UNEXPECTED EXCEPTION: NameError("name 'outputs' is not defined")
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest transformers.models.albert.modeling_albert.AlbertForMaskedLM.forward[12]>", line 1, in <module>
NameError: name 'outputs' is not defined
/home/vumichien/Detomo/transformers/src/transformers/models/albert/modeling_albert.py:1046: UnexpectedException
=================================================================================== warnings summary ===================================================================================
venv/lib/python3.8/site-packages/flatbuffers/compat.py:19
  /home/vumichien/Detomo/transformers/venv/lib/python3.8/site-packages/flatbuffers/compat.py:19: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================== short test summary info ================================================================================
FAILED src/transformers/models/albert/modeling_albert.py::transformers.models.albert.modeling_albert.AlbertForMaskedLM.forward
======================================================================= 1 failed, 6 passed, 1 warning in 52.18s ========================================================================
``

**Comment by ydshieh** [comment]
Maybe a quick easy way is just to overwrite the examples for AlbertForMaskedLM in the model files. Something similar to https://github.com/huggingface/transformers/pull/16565#discussion_r843972010

But that case is reversed: masked input has fewer tokens. So you need to have some different operations. 

Let's wait @patrickvonplaten to see if he has better suggestion.

**Review by sgugger — APPROVED** [review]
Good for me once the Albert example for masked language modeling is fixed. Thanks!

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 1]
```diff
@@ -2397,6 +2397,8 @@ def set_output_embeddings(self, new_embeddings):
         checkpoint=_CHECKPOINT_FOR_DOC,
         output_type=MaskedLMOutput,
         config_class=_CONFIG_FOR_DOC,
+        expected_output="'here'",
```
Comment (patrickvonplaten) [inline src/transformers/models/big_bird/modeling_big_bird.py:2400]: Sorry to comment so late here. Could we maybe overwrite the BigBird example as well? https://huggingface.co/google/bigbird-roberta-base has quite a significant number of downloads and it's know to be a **long-range** model. Could we maybe provide a long input to be masked here and to all other examples below as well? 

Would be great if we could overwrite the example doc string here @ydshieh

**Comment by patrickvonplaten** [comment]
@vumichien @ydshieh, I'd be in favor of overwriting both Albert (so that MLM is correct) as well as BigBird (to show that it's long-range). What do you think?

File: src/transformers/models/big_bird/modeling_big_bird.py (hunk 1 above)
Comment (vumichien) [reply src/transformers/models/big_bird/modeling_big_bird.py:2400]: I think it's a great idea. Let me prepare better examples for Bigbird.

**Comment by vumichien** [comment]
@ydshieh @patrickvonplaten I have overwritten both the doc-test examples of Albert and Bigbird. What do you think about them?

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 2]
```diff
@@ -2420,6 +2413,59 @@ def forward(
...
+
+        ```python
+        >>> import torch
+        >>> from transformers import BigBirdTokenizer, BigBirdForMaskedLM
+
+        >>> tokenizer = BigBirdTokenizer.from_pretrained("google/bigbird-roberta-base")
+        >>> model = BigBirdForMaskedLM.from_pretrained("google/bigbird-roberta-base")
+
+        >>> # add mask_token
+        >>> ARTICLE_TO_MASK = (
```
Comment (patrickvonplaten) [inline src/transformers/models/big_bird/modeling_big_bird.py:2429]:
```suggestion
        >>> LONG_ARTICLE_TO_MASK = (
```

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 3]
```diff
@@ -2420,6 +2413,59 @@ def forward(
...
+        ...     "BIGBIRD satisfies a number of theoretical results: it is a universal approximator of sequence to "
+        ...     "sequence functions and is also Turing complete. Theoretically, authors use the power of extra "
+        ...     "global tokens preserve the expressive powers of the model. Authors complement these results by "
+        ...     "showing that moving to sparse attention mechanism do incur a cost. Empirically, BIGBIRD gives "
+        ...     "state-of-the-art performance on a number of NLP tasks such as question [MASK] and long document "
+        ...     "classification. Authors further introduce attention based contextual language model for DNA and "
+        ...     "fine-tune it for down stream tasks such as promoter region prediction and predicting effects of "
+        ...     "non-coding variants."
+        ... )
+        >>> inputs = tokenizer(ARTICLE_TO_MASK, return_tensors="pt")
```
Comment (patrickvonplaten) [inline src/transformers/models/big_bird/modeling_big_bird.py:2440]:
```suggestion
        >>> inputs = tokenizer(LONG_ARTICLE_TO_MASK, return_tensors="pt")
```

Comment (patrickvonplaten) [reply src/transformers/models/big_bird/modeling_big_bird.py:2440]: Just to make it very clear that BigBird can do long-range ;-)

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 4]
```diff
@@ -2677,6 +2716,42 @@ def forward(
...
+        Example:
+
+        ```python
+        >>> import torch
+        >>> from transformers import BigBirdTokenizer, BigBirdForSequenceClassification
+
+        >>> tokenizer = BigBirdTokenizer.from_pretrained("l-yohai/bigbird-roberta-base-mnli")
+        >>> model = BigBirdForSequenceClassification.from_pretrained("l-yohai/bigbird-roberta-base-mnli")
+
+        >>> text = (
```
Comment (patrickvonplaten) [inline src/transformers/models/big_bird/modeling_big_bird.py:2731]:
```suggestion
        >>> text = 50 * (
```
to make the article really long? :-)

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 5]
```diff
@@ -2983,6 +3049,47 @@ def forward(
...
+        >>> question = "What is the oldest structure at Notre Dame?"
+        >>> inputs = tokenizer(question, text, return_tensors="pt")
+        >>> with torch.no_grad():
+        ...     outputs = model(**inputs)
+
+        >>> answer_start_index = outputs.start_logits.argmax()
+        >>> answer_end_index = outputs.end_logits.argmax()
+        >>> predict_answer_tokens = inputs.input_ids[0, answer_start_index : answer_end_index + 1]
+        >>> tokenizer.decode(predict_answer_tokens)
+        'Old College'
```
Comment (patrickvonplaten) [inline src/transformers/models/big_bird/modeling_big_bird.py:3083]: Very cool example!

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 6]
```diff
@@ -2983,6 +3049,47 @@ def forward(
...
+        Example:
+
+        ```python
+        >>> import torch
+        >>> from transformers import BigBirdTokenizer, BigBirdForQuestionAnswering
+
+        >>> tokenizer = BigBirdTokenizer.from_pretrained("abhinavkulkarni/bigbird-roberta-base-finetuned-squad")
+        >>> model = BigBirdForQuestionAnswering.from_pretrained("abhinavkulkarni/bigbird-roberta-base-finetuned-squad")
+
+        >>> text = (
```
Comment (patrickvonplaten) [inline src/transformers/models/big_bird/modeling_big_bird.py:3064]:
```suggestion
        >>> text = 50 * (
```

**Comment by vumichien** [comment]
I have changed the longer examples for doctest. The examples are quite long, but in my opinion,  they are good to show that Bigbird is long-range model

**Comment by sgugger** [comment]
Can we put that text in some dataset instead? The documentation will become a bit unreadable with such a long text, where as we could just load a dataset in one line and take the first sample.

**Comment by vumichien** [comment]
@sgugger Thank you for your suggestion. I have changed to use the examples from squad datasets. How do you think about that?

**Comment by sgugger** [comment]
Way better, and great that you're showing the shape! Good for me if @patrickvonplaten is okay.

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 7]
```diff
@@ -2418,6 +2413,44 @@ def forward(
...
+        >>> from transformers import BigBirdTokenizer, BigBirdForMaskedLM
+        >>> from datasets import load_dataset
+
+        >>> tokenizer = BigBirdTokenizer.from_pretrained("google/bigbird-roberta-base")
+        >>> model = BigBirdForMaskedLM.from_pretrained("google/bigbird-roberta-base")
+        >>> squad_ds = load_dataset("squad_v2", split="train")  # doctest: +IGNORE_RESULT
+
+        >>> LONG_ARTICLE_TARGET = squad_ds[81514]["context"]
+        >>> # add mask_token
+        >>> LONG_ARTICLE_TO_MASK = LONG_ARTICLE_TARGET.replace("maximum", "[MASK]")
```
Comment (ydshieh) [inline src/transformers/models/big_bird/modeling_big_bird.py:2432]: just a tiny nit: could we show a few words around the target world `maximum`? Just for the readers to be able to see the context, and find the output indeed make sense :)

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 8]
```diff
@@ -2858,9 +2910,12 @@ def __init__(self, config):
     @add_start_docstrings_to_model_forward(BIG_BIRD_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
     @add_code_sample_docstrings(
         processor_class=_TOKENIZER_FOR_DOC,
-        checkpoint=_CHECKPOINT_FOR_DOC,
+        checkpoint="vumichien/token-classification-bigbird-roberta-base",
```
Comment (ydshieh) [inline src/transformers/models/big_bird/modeling_big_bird.py:2913]: Wow, you have trained a token classification model ..? 💯

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 9]
```diff
@@ -2985,6 +3035,44 @@ def forward(
...
+        >>> import torch
+        >>> from transformers import BigBirdTokenizer, BigBirdForQuestionAnswering
+        >>> from datasets import load_dataset
+
+        >>> tokenizer = BigBirdTokenizer.from_pretrained("abhinavkulkarni/bigbird-roberta-base-finetuned-squad")
+        >>> model = BigBirdForQuestionAnswering.from_pretrained("abhinavkulkarni/bigbird-roberta-base-finetuned-squad")
+        >>> squad_ds = load_dataset("squad_v2", split="train")  # doctest: +IGNORE_RESULT
+
+        >>> LONG_ARTICLE = squad_ds[81514]["context"]
+        >>> QUESTION = squad_ds[81514]["question"]
```
Comment (ydshieh) [inline src/transformers/models/big_bird/modeling_big_bird.py:3053]: nit: would be good to show the question text as a comment

**Review by ydshieh — APPROVED** [review]
As your previous PR, very high quality!
Thank you @vumichien for the effort to overwrite the doctest code 💯 

I leave 2 tiny suggestions, but no need to feel necessary to do it.

Ran locally -> all tests passed!

File: src/transformers/models/big_bird/modeling_big_bird.py (hunk 9 above)
Comment (vumichien) [reply src/transformers/models/big_bird/modeling_big_bird.py:3059]: I will revise as your suggestion

File: src/transformers/models/big_bird/modeling_big_bird.py (hunk 7 above)
Comment (vumichien) [reply src/transformers/models/big_bird/modeling_big_bird.py:2437]: I will revise as your suggestion

File: src/transformers/models/big_bird/modeling_big_bird.py (hunk 8 above)
Comment (vumichien) [reply src/transformers/models/big_bird/modeling_big_bird.py:2918]: Sorry, it's just a random weight...(but maybe I will try to train bigbird for token classification in the near future 😅). The reason why I didn't use a model from `hf-internal-testing` (hf-internal-testing/tiny-random-bigbird_pegasus) is that I think it's also the random weight model but the output is too long. But if you think it's not a good way, I will revise with this checkpoint `hf-internal-testing/tiny-random-bigbird_pegasus`

**Comment by vumichien** [comment]
@ydshieh I have revised as your suggestion. Please let me know if I need to revise something.

File: src/transformers/models/big_bird/modeling_big_bird.py (hunk 8 above)
Comment (ydshieh) [reply src/transformers/models/big_bird/modeling_big_bird.py:2918]: It's OK, it is completely fine.

However, if `vumichien/token-classification-bigbird-roberta-base` has random weights, it is a good idea to have the name like

```
vumichien/token-classification-bigbird-roberta-base-random
```

This way, the doc reader and hub users won't be confused 😄

**Comment by ydshieh** [comment]
> @ydshieh I have revised as your suggestion. Please let me know if I need to revise something.

Love it! Thank you.
I will let @patrickvonplaten to have a final look (if any) & click the merge button 💯

File: src/transformers/models/big_bird/modeling_big_bird.py (hunk 8 above)
Comment (vumichien) [reply src/transformers/models/big_bird/modeling_big_bird.py]: Ah, I see. I will change the checkpoint name

File: src/transformers/models/big_bird/modeling_big_bird.py [hunk 10]
```diff
@@ -2686,6 +2706,43 @@ def forward(
...
+        Example:
+
+        ```python
+        >>> import torch
+        >>> from transformers import BigBirdTokenizer, BigBirdForSequenceClassification
+        >>> from datasets import load_dataset
+
+        >>> tokenizer = BigBirdTokenizer.from_pretrained("l-yohai/bigbird-roberta-base-mnli")
+        >>> model = BigBirdForSequenceClassification.from_pretrained("l-yohai/bigbird-roberta-base-mnli")
+        >>> squad_ds = load_dataset("squad_v2", split="train")  # doctest: +IGNORE_RESULT
```
Comment (patrickvonplaten) [inline src/transformers/models/big_bird/modeling_big_bird.py:2721]: That's great!

**Review by patrickvonplaten — APPROVED** [review]
Amazing job @vumichien - thanks a mille for making the example so nice :-)

**Comment by ydshieh** [comment]
Running the last time -> will merge if all tests pass

**Pull request merged**